applications/FluidDynamicsApplication/custom_utilities/fluid_calculation_utilities.cpp

```diff
@@ -65,6 +65,159 @@ void FluidCalculationUtilities::UpdateValue(
     noalias(rOutput) += rInput;
 }
 
+template<>
+void FluidCalculationUtilities::AssignGradientValue(
+    const double& rInput,
+    const double rShapeFunctionDerivative,
+    const IndexType DirectionIndex,
+    array_1d<double, 3>& rOutput)
+{
+    KRATOS_TRY
+
```

Comment:
```suggestion
    KRATOS_DEBUG_ERROR_IF(DirectionIndex > 2)
        << "Unsupported Direction Index. DirectionIndex should be  less than 3 "
           "[ DirectionIndex = "
        << DirectionIndex << " ].\n";
```

Shouldn't be a direction check here too?

Reply:
got rid of these template specializations, now it is more simpler with `std::apply`